src/python/pants/backend/jvm/tasks/classmap.py

```diff
@@ -0,0 +1,51 @@
+# coding=utf-8
+# Copyright 2016 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from pants.backend.jvm.targets.jar_library import JarLibrary
+from pants.backend.jvm.tasks.classpath_util import ClasspathUtil
+from pants.task.console_task import ConsoleTask
+
+
+class ClassmapTask(ConsoleTask):
+  """Print a mapping from class name to the owning target from target's runtime classpath."""
+
+  @classmethod
+  def register_options(cls, register):
+    super(ClassmapTask, cls).register_options(register)
+
+    register('--internal-only', default=False, type=bool, fingerprint=True,
+             help='Specifies that only class names of internal dependencies should be included.')
+    register('--transitive', default=True, type=bool,
+             help='Score all targets in the build graph transitively.')
+
+  def classname_for_classfile(self, target, classpath_products):
+    contents = ClasspathUtil.classpath_contents((target,), classpath_products)
+    for f in contents:
+      classname = ClasspathUtil.classname_for_rel_classfile(f)
+      # None for non `.class` files
+      if classname:
+        yield classname
+
+  def console_output(self, _):
+    def should_ignore(target):
+      if target.address.spec.startswith('//:'):
```

Comment:
Removed the `//:` check, agree easy to filter, either before (if calling with `--internal-only`) or after on the output.

The extra benefit is to be honest allow us to inspect all dependencies including those from build tools.